qiskit/circuit/parametertable.py

```diff
@@ -27,7 +29,10 @@ def __init__(self, *args, **kwargs):
         the structure of _table is,
            {var_object: [(instruction_object, parameter_index), ...]}
         """
-        self._table = dict(*args, **kwargs)
+        if sys.version_info.major >= 3 and sys.version_info.minor >= 6:
```

Comment:
For readability, could we use `OrderedDict` throughout (and then replace with dict as needed when python 3.5 reaches EOL)?

Reply:
That sounds good. Probably also improves readability since it’s obviously ordered. 
And even if there were any performance differences it probably wouldn’t matter since the number of parameters isn’t excessively large.